### PR TITLE
platformio 2.8.4 (new formula)

### PR DESCRIPTION
**Pull request by ivankravets**
Requested by https://github.com/platformio/platformio/issues/395

# Audit
```
ikravets@innomac ~> brew audit --online --strict platformio
==> brew style platformio

1 file inspected, no offenses detected
```

### CODE REVIEW

**Comment by DomT4** [comment]
```
==> audit problems
platformio:
 * 58: Trailing whitespace was found
```
:wink:

File: Library/Formula/platformio.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  resource "requests" do
+    url "https://pypi.python.org/packages/source/r/requests/requests-2.9.1.tar.gz"
+    sha256 "c577815dd00f1394203fc44eb979724b098f88264a9ef898ee45b8e5e9cf587f"
+  end
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    %w[bottle click colorama lockfile pyserial requests].each do |r|
```
Comment (DomT4) [inline Library/Formula/platformio.rb]: You can use:
```ruby
resources.each do |r|
  r.stage do
```
here. It's a bit cleaner and fine when there's no extra handling required.

File: Library/Formula/platformio.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    system "platformio"
```
Comment (DomT4) [inline Library/Formula/platformio.rb]: Please call these with a full path, i.e. `system bin/"platformio"`. Ideally the test would be stronger on probing functionality, if at all possible.

**Comment by ivankravets** [comment]
@DomT4 Thanks. Please check https://github.com/ivankravets/homebrew/commit/0f7e711c9ac599c2530f86c030f1b4cf95577f61

File: Library/Formula/platformio.rb (hunk 2 above)
Comment (ivankravets) [inline Library/Formula/platformio.rb]: > Ideally the test would be stronger on probing functionality, if at all possible.

`platformio` returns HELP screen by default and loads all functionality. If `exit code` is equal to `0`, then all are installed fine.

**Comment by sagikazarmark** [comment]
:+1:

**Comment by DomT4** [comment]
Seems fine to me. Merged in https://github.com/Homebrew/homebrew/commit/e8d9939b43f9b598ba727647df78ac588517f5de. Thank you for your contribution to Homebrew @ivankravets; we appreciate it! :smiley_cat:

**Comment by ivankravets** [comment]
Thanks guys! https://twitter.com/platformio_org/status/704287776490004480

@DomT4 should I make new PR each time when PlatformIO is updated? Could we automate this process and use the latest version from https://pypi.python.org/pypi/platformio/ ?

PyPi provides whole required information https://pypi.python.org/pypi/platformio/json

**Comment by DomT4** [comment]
> should I make new PR each time when PlatformIO is updated?

You or anyone else who wants to :smiley:.

> Could we automate this process and use the latest version from https://pypi.python.org/pypi/platformio/ ?

We prefer human verification for various reasons, but you're more than welcome to set up a script or bot that submits PRs for the project when new versions are available as long as the account is monitored enough to respond to queries if something breaks.

**Comment by ivankravets** [comment]
I mean do we have ability to create "platformio.rb" which automatically monitors updates?

**Comment by DomT4** [comment]
At the moment there's no mechanism that notifies users/contributors/maintainers a package has an update available, no. Various people have taken a swing at that problem over the years but we haven't found a consistent solution yet.

**Comment by ivankravets** [comment]
@DomT4 thanks a lot! I'll make PR for the new version.

Have a nice day!

**Comment by DomT4** [comment]
Thanks @ivankravets, you too!

**Pull request closed without merging**